pkg/api/tasks.go

```diff
@@ -0,0 +1,178 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"go1f/pkg/db"
+	"io"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+)
+
+type TasksResp struct {
+	Tasks []map[string]string `json:"tasks"`
+}
+
+func writeJson(w http.ResponseWriter, data interface{}) {
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(data); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+}
+
+func writeError(w http.ResponseWriter, status int, message string) {
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(status)
+	json.NewEncoder(w).Encode(map[string]string{"error": message})
+}
+func TasksHandler(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case http.MethodGet:
+		db.GetTaskHandler(w, r)
+	case http.MethodPost:
+		AddTaskHandler(w, r)
+	case http.MethodPut:
+		UpdateTaskHandler(w, r)
+	case http.MethodDelete:
+		DeleteTaskHandler(w, r)
+	default:
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		writeJson(w, map[string]string{"error": "Метод не разрешен"})
+	}
+}
+
+func ListTasksHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodGet {
+		writeError(w, http.StatusMethodNotAllowed, "Метод не разрешен")
+		return
+	}
+	tasks, err := db.Tasks(50)
+	if err != nil {
+		writeError(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+	writeJson(w, map[string]interface{}{"tasks": tasks})
+}
+
+// PUT /api/task
+func UpdateTaskHandler(w http.ResponseWriter, r *http.Request) {
+	bodyBytes, err := io.ReadAll(r.Body)
+	if err != nil {
+		writeJson(w, map[string]string{"error": "Ошибка чтения тела запроса"})
+		return
+	}
+	defer r.Body.Close()
+
+	bodyStr := string(bodyBytes)
+	bodyStr = strings.ReplaceAll(bodyStr, `"repeat":[]`, `"repeat":""`)
+
+	var task db.Task
+	if err := json.Unmarshal([]byte(bodyStr), &task); err != nil {
+		writeJson(w, map[string]string{"error": "Неверный формат JSON"})
+		return
+	}
+
+	if task.ID == "" {
+		writeJson(w, map[string]string{"error": "Не указан идентификатор"})
+		return
+	}
+	if task.Date == "" || task.Title == "" {
+		writeJson(w, map[string]string{"error": "Поля date и title обязательны"})
+		return
+	}
+
+	if err := validateTask(task.Date, task.Title, task.Repeat); err != nil {
+		writeJson(w, map[string]string{"error": err.Error()})
+		return
+	}
+
+	if err := db.UpdateTask(&task); err != nil {
+		writeJson(w, map[string]string{"error": err.Error()})
+		return
+	}
+
+	writeJson(w, map[string]interface{}{})
+}
+
+func validateTask(date, title, repeat string) error {
+	if len(date) != 8 {
+		return fmt.Errorf("Дата должна быть в формате YYYYMMDD")
+	}
+	d, err := time.Parse("20060102", date)
+	if err != nil {
+		return fmt.Errorf("Неверный формат даты")
```

Comment:
1. Теряешь сообщение об ошибки
2. Ошибку лучше писать на английском
3. Ошибку лучше начинать с маленькой буквы
```suggestion
		return fmt.Errorf("wrong date format: %w", err)
```